centreon/packages/ui/src/Graph/BasicComponents/Lines/index.tsx

```diff
@@ -0,0 +1,232 @@
+import { MutableRefObject } from 'react';
+
+import { ScaleLinear } from 'd3-scale';
+import { isEmpty, isNil } from 'ramda';
+
+import RegularAnchorPoint from '../../InteractiveComponents/AnchorPoint/RegularAnchorPoint';
+import { displayArea } from '../../helpers/index';
+import { GlobalAreaLines } from '../../models';
+import { getStackedYScale, getUnits, getYScale } from '../../timeSeries';
+import { Line, TimeValue } from '../../timeSeries/models';
+
+import RegularLine from './RegularLines';
+import useRegularLines from './RegularLines/useRegularLines';
+import StackedLines from './StackedLines';
+import useStackedLines from './StackedLines/useStackedLines';
+import ThresholdLines from './Threshold';
+import AwesomeCircles from './Threshold/Circle';
+import ThresholdWithPatternLines from './Threshold/ThresholdWithPatternLines';
+import ThresholdWithVariation from './Threshold/ThresholdWithVariation';
+import { Data } from './Threshold/models';
+import useDataThreshold from './Threshold/useDataThreshold';
+
+interface Props extends GlobalAreaLines {
+  displayAnchor: boolean;
+  displayedLines: Array<Line>;
+  graphSvgRef: MutableRefObject<SVGSVGElement | null>;
+  height: number;
+  leftScale: ScaleLinear<number, number>;
+  rightScale: ScaleLinear<number, number>;
+  timeSeries: Array<TimeValue>;
+  width: number;
+  xScale: ScaleLinear<number, number>;
+}
+
+const Lines = ({
+  height,
+  graphSvgRef,
+  width,
+  displayAnchor,
+  leftScale,
+  rightScale,
+  xScale,
+  timeSeries,
+  displayedLines,
+  areaThresholdLines,
+  areaStackedLines,
+  areaRegularLines
+}: Props): JSX.Element => {
+  const { stackedLinesData, invertedStackedLinesData } = useStackedLines({
+    lines: displayedLines,
+    timeSeries
+  });
+
+  const { regularLines } = useRegularLines({ lines: displayedLines });
+
+  const displayEnvelopeThreshold =
+    !isNil(areaThresholdLines?.factors?.currentFactorMultiplication) &&
+    !isNil(areaThresholdLines?.factors?.simulatedFactorMultiplication);
+
+  const displayCircles = areaThresholdLines?.displayCircles ?? true;
+
+  const currentFactorMultiplication = areaThresholdLines?.factors
+    ?.currentFactorMultiplication as number;
+
+  const simulatedFactorMultiplication = areaThresholdLines?.factors
```

Comment:
What do you think to split `areaThresholdLines` into several type of thresholdLines. And each type have their own prop.

```
{
  "areaThresholdLines": [
    {
      "type": "variation",
      "getX": ...,
      "getY0": ...,
      "getY1": ...,
      "lineColorY0": ...,
      "lineColorY1": ...
    },
    {
      "type": "pattern",
      "orientation": ["diagonal"],
      ...
    },
    ...
  ]
}
```

And you can also extract the logic of `getVariationEnvelopThresholdData` from the Graph implementation in order to be use externally (in the component in which the component is implemented).

Reply:
for the thresholds they all have the same getY0, getY1, getX the functions which calculate the points P, P1, P0 .. because it is the same data, except for threshold variation , which we must add a variation to the same points which exist when calculating points, the formula used is dependent on the points that already exist, for the user he is only able to provide the indicators, for this I created a cutom hook which returns the shared functions and in the case of the threshold type variation will return more functions for the variation points 🤔  🤔